File: youlexuan-sellergoods-interface/src/main/java/com/offcn/sellergoods/service/SpecificationService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.entity.SpecificationVO;
import com.offcn.pojo.TbSpecification;

import java.util.List;

public interface SpecificationService {
    /**
     * 查询所有的规格【不带条件不分页】
     * @return
     */
    public List<TbSpecification> findAll();

    /**
     * 查询所有的规格【带分页带条件】
     * @param pageNum
     * @param pageSize
     * @param tbSpecification
     * @return
     */
    public PageResult findPage(int pageNum,int pageSize,TbSpecification tbSpecification);
    /**
     * 添加规格
     * @param specificationVO
     */
    public void add(SpecificationVO specificationVO);

    /**
     * 根据id获取组合实体
     * @param id
     * @return
     */
    public SpecificationVO findOne(Long id);
    /**
     * 更新规格
     * @param specificationVO
     */
    public void update(SpecificationVO specificationVO);

    /**
     * 批量删除规格
     * @param ids
     */
    public void delete(Long[] ids);
}

File: youlexuan-shop-web/src/main/webapp/js/service/specificationService.js
app.service('specificationService', function ($http) {
    this.findAll = function () {
        return $http.get('../specification/findAll.do');
    }

    this.add = function (entity) {
        return $http.post('../specification/add.do', entity);
    }

    this.findPage = function (pageNum, pageSize, entity) {
        return $http.post('../specification/findPage.do?pageNum=' + pageNum + '&pageSize=' + pageSize, entity);
    }

    this.findOne = function (id) {
        return $http.get('../specification/findOne.do?id='+id);
    }

    this.update = function (entity) {
        return $http.post('../specification/update.do',entity);
    }

    this.delete = function (ids) {
        return $http.get('../specification/delete.do?ids='+ids);
    }

})